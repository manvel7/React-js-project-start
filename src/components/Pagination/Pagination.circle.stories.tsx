import React, {useState} from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Pagination, {IProps} from "./Pagination"

const meta: Meta<IProps> = {
  title: "Components/Pagination/Circle",
  component: Pagination,
  argTypes: {}
}

const PaginationStory: Story<IProps> = (args) => {
  const {currentPage, ...rest} = args
  const [page, setPage] = useState<number>(currentPage)

  const onChange = (page: number) => {
    setPage(page)
  }

  return (
    <Pagination
      {...rest}
      variant="circle"
      currentPage={page}
      onPageChange={onChange}
    />
  )
}

const Medium = PaginationStory.bind({})
Medium.args = {
  pageCount: 7,
  currentPage: 1
}

const Large = PaginationStory.bind({})
Large.args = {
  size: "lg",
  pageCount: 3,
  currentPage: 1
}

const WithoutNumbers = PaginationStory.bind({})
WithoutNumbers.args = {
  pageCount: 3,
  currentPage: 1,
  hidePageNumber: true
}

export {Medium, Large, WithoutNumbers}

export default meta
