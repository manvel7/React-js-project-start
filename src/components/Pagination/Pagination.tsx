import React, {FunctionComponent, HTMLAttributes, useMemo} from "react"
import {faAngleLeft, faAngleRight} from "@fortawesome/pro-solid-svg-icons"
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome"
import clsx from "clsx"
import _ from "lodash"
import Button from "../Button"
import PaginationButton from "./Button"
import {ButtonVariant} from "./Button/PaginationButton"
import PaginationWrapper from "./PaginationWrapper"

export type PaginationSize = "md" | "lg"

export interface IProps extends HTMLAttributes<HTMLDivElement> {
  size?: PaginationSize
  currentPage: number
  pageCount: number
  onPageChange?: (page: number) => void
  variant?: ButtonVariant
  hidePageNumber?: boolean
}

const HALF_RANGE = 2
const FIRST_PAGES_COUNT = 5
const LAST_PAGES_OFFSET = 4
const ELLIPSIS_OFFSET = 4
const FIRST_LAST_PAGE_OFFSET = 3
const ELLIPSIS_SKIP = 5
const FIRST_LAST_PAGE_VISIBLE = 5

const Pagination: FunctionComponent<IProps> = (props) => {
  const {
    currentPage,
    pageCount,
    onPageChange,
    size,
    variant,
    hidePageNumber
  } = props

  const onPreviousClick = () => {
    if (onPageChange) {
      onPageChange(currentPage - 1)
    }
  }

  const onNextClick = () => {
    if (onPageChange) {
      onPageChange(currentPage + 1)
    }
  }

  const setPage = (page: number) => () => {
    if (onPageChange) {
      if (page < 1) {
        onPageChange(1)
      } else if (page > pageCount) {
        onPageChange(pageCount)
      } else {
        onPageChange(page)
      }
    }
  }

  const pages = useMemo<number[]>(() => {
    if (currentPage <= HALF_RANGE) {
      return _.range(1, _.min([FIRST_PAGES_COUNT + 1, pageCount + 1]))
    }

    if (currentPage > pageCount - HALF_RANGE) {
      return _.range(
        _.max([1, pageCount - LAST_PAGES_OFFSET]) || 1,
        pageCount + 1
      )
    }

    return _.range(currentPage - HALF_RANGE, currentPage + HALF_RANGE + 1)
  }, [currentPage, pageCount])

  return (
    <PaginationWrapper>
      <Button
        className={clsx({hidden: currentPage === 1})}
        onClick={onPreviousClick}
        variant="text"
        size={size}>
        <FontAwesomeIcon icon={faAngleLeft} style={{marginRight: "0.8rem"}} />
        Previous
      </Button>
      {pageCount > FIRST_LAST_PAGE_VISIBLE && (
        <>
          {currentPage > FIRST_LAST_PAGE_OFFSET && (
            <PaginationButton
              onClick={setPage(1)}
              size={size}
              variant={variant}>
              1
            </PaginationButton>
          )}
          {currentPage > ELLIPSIS_OFFSET && (
            <PaginationButton
              onClick={setPage(currentPage - ELLIPSIS_SKIP)}
              size={size}
              variant={variant}>
              &hellip;
            </PaginationButton>
          )}
        </>
      )}
      {pages.map((page) => (
        <PaginationButton
          key={page}
          isActive={page === currentPage}
          disabled={page === currentPage}
          onClick={setPage(page)}
          size={size}
          variant={variant}>
          {!hidePageNumber && page}
        </PaginationButton>
      ))}
      {pageCount > FIRST_LAST_PAGE_VISIBLE && (
        <>
          {currentPage <= pageCount - ELLIPSIS_OFFSET && (
            <PaginationButton
              onClick={setPage(currentPage + ELLIPSIS_SKIP)}
              size={size}
              variant={variant}>
              &hellip;
            </PaginationButton>
          )}
          {currentPage <= pageCount - FIRST_LAST_PAGE_OFFSET && (
            <PaginationButton
              onClick={setPage(pageCount)}
              size={size}
              variant={variant}>
              {pageCount}
            </PaginationButton>
          )}
        </>
      )}
      <Button
        className={clsx({hidden: currentPage === pageCount})}
        onClick={onNextClick}
        variant="text"
        size={size}>
        Next
        <FontAwesomeIcon icon={faAngleRight} style={{marginLeft: "0.8rem"}} />
      </Button>
    </PaginationWrapper>
  )
}

Pagination.defaultProps = {
  size: "md",
  variant: "normal",
  hidePageNumber: false
}

export default Pagination
