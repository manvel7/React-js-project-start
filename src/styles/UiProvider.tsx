import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
