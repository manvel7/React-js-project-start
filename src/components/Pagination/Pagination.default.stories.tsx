import React, {useState} from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Pagination, {IProps} from "./Pagination"

const meta: Meta<IProps> = {
  title: "Components/Pagination/Default",
  component: Pagination,
  argTypes: {}
}

const PaginationStory: Story<IProps> = (args) => {
  const {currentPage, ...rest} = args
  const [page, setPage] = useState<number>(currentPage)

  const onChange = (page: number) => {
    setPage(page)
  }

  return <Pagination {...rest} currentPage={page} onPageChange={onChange} />
}

const Medium = PaginationStory.bind({})
Medium.args = {
  pageCount: 17,
  currentPage: 1
}

const Large = PaginationStory.bind({})
Large.args = {
  size: "lg",
  pageCount: 11,
  currentPage: 1
}

export {Medium, Large}

export default meta
