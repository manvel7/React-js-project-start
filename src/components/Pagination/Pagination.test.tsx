import React from "react"
import {render} from "@testing-library/react"
import {UiProvider} from "../../styles"
import Pagination from "./Pagination"

describe("Pagination", () => {
  describe("Default variant", () => {
    describe("with medium size", () => {
      it("should match snapshot", () => {
        const result = render(
          <UiProvider>
            <Pagination size="md" pageCount={3} currentPage={1} />
          </UiProvider>
        )
        expect(result.container.firstChild).toMatchSnapshot()
      })
    })

    describe("with large size", () => {
      it("should match snapshot", () => {
        const result = render(
          <UiProvider>
            <Pagination size="lg" pageCount={3} currentPage={1} />
          </UiProvider>
        )
        expect(result.container.firstChild).toMatchSnapshot()
      })
    })
  })

  describe("Circle variant", () => {
    describe("with medium size", () => {
      it("should match snapshot", () => {
        const result = render(
          <UiProvider>
            <Pagination
              variant="circle"
              size="md"
              pageCount={3}
              currentPage={1}
            />
          </UiProvider>
        )
        expect(result.container.firstChild).toMatchSnapshot()
      })
    })

    describe("with large size", () => {
      it("should match snapshot", () => {
        const result = render(
          <UiProvider>
            <Pagination
              variant="circle"
              size="lg"
              pageCount={3}
              currentPage={1}
            />
          </UiProvider>
        )
        expect(result.container.firstChild).toMatchSnapshot()
      })
    })

    describe("without numbers", () => {
      it("should match snapshot", () => {
        const result = render(
          <UiProvider>
            <Pagination
              variant="circle"
              hidePageNumber={true}
              pageCount={3}
              currentPage={1}
            />
          </UiProvider>
        )
        expect(result.container.firstChild).toMatchSnapshot()
      })
    })
  })
})
