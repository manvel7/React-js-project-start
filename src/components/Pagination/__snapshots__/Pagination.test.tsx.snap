// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Circle variant with large size should match snapshot 1`] = `
.c1 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.8rem;
  padding: 0.8rem 1.6rem;
  font-size: 2.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: transparent;
  color: #001254;
}

.c1 b,
.c1 strong {
  font-weight: 800;
}

.c2 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 2.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c2[disabled] {
  cursor: initial;
}

.c2:first-child {
  margin-left: 0;
}

.c2:last-child {
  margin-right: 0;
}

.c2 b,
.c2 strong {
  font-weight: 800;
}

.c3 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 2.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
}

.c3[disabled] {
  cursor: initial;
}

.c3:first-child {
  margin-left: 0;
}

.c3:last-child {
  margin-right: 0;
}

.c3 b,
.c3 strong {
  font-weight: 800;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="c0"
>
  <button
    class="c1 hidden"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-left fa-w-8 "
      data-icon="angle-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        fill="currentColor"
      />
    </svg>
    Previous
  </button>
  <button
    class="c2"
    disabled=""
  >
    1
  </button>
  <button
    class="c3"
  >
    2
  </button>
  <button
    class="c3"
  >
    3
  </button>
  <button
    class="c1"
  >
    Next
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-right fa-w-8 "
      data-icon="angle-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-left: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination Circle variant with medium size should match snapshot 1`] = `
.c1 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
}

.c1 b,
.c1 strong {
  font-weight: 800;
}

.c2 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c2[disabled] {
  cursor: initial;
}

.c2:first-child {
  margin-left: 0;
}

.c2:last-child {
  margin-right: 0;
}

.c2 b,
.c2 strong {
  font-weight: 800;
}

.c3 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
}

.c3[disabled] {
  cursor: initial;
}

.c3:first-child {
  margin-left: 0;
}

.c3:last-child {
  margin-right: 0;
}

.c3 b,
.c3 strong {
  font-weight: 800;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="c0"
>
  <button
    class="c1 hidden"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-left fa-w-8 "
      data-icon="angle-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        fill="currentColor"
      />
    </svg>
    Previous
  </button>
  <button
    class="c2"
    disabled=""
  >
    1
  </button>
  <button
    class="c3"
  >
    2
  </button>
  <button
    class="c3"
  >
    3
  </button>
  <button
    class="c1"
  >
    Next
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-right fa-w-8 "
      data-icon="angle-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-left: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination Circle variant without numbers should match snapshot 1`] = `
.c1 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
}

.c1 b,
.c1 strong {
  font-weight: 800;
}

.c2 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c2[disabled] {
  cursor: initial;
}

.c2:first-child {
  margin-left: 0;
}

.c2:last-child {
  margin-right: 0;
}

.c2 b,
.c2 strong {
  font-weight: 800;
}

.c3 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 10rem;
}

.c3[disabled] {
  cursor: initial;
}

.c3:first-child {
  margin-left: 0;
}

.c3:last-child {
  margin-right: 0;
}

.c3 b,
.c3 strong {
  font-weight: 800;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="c0"
>
  <button
    class="c1 hidden"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-left fa-w-8 "
      data-icon="angle-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        fill="currentColor"
      />
    </svg>
    Previous
  </button>
  <button
    class="c2"
    disabled=""
  />
  <button
    class="c3"
  />
  <button
    class="c3"
  />
  <button
    class="c1"
  >
    Next
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-right fa-w-8 "
      data-icon="angle-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-left: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination Default variant with large size should match snapshot 1`] = `
.c1 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.8rem;
  padding: 0.8rem 1.6rem;
  font-size: 2.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: transparent;
  color: #001254;
}

.c1 b,
.c1 strong {
  font-weight: 800;
}

.c2 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 2.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 0.4rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c2[disabled] {
  cursor: initial;
}

.c2:first-child {
  margin-left: 0;
}

.c2:last-child {
  margin-right: 0;
}

.c2 b,
.c2 strong {
  font-weight: 800;
}

.c3 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 2.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 0.4rem;
}

.c3[disabled] {
  cursor: initial;
}

.c3:first-child {
  margin-left: 0;
}

.c3:last-child {
  margin-right: 0;
}

.c3 b,
.c3 strong {
  font-weight: 800;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="c0"
>
  <button
    class="c1 hidden"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-left fa-w-8 "
      data-icon="angle-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        fill="currentColor"
      />
    </svg>
    Previous
  </button>
  <button
    class="c2"
    disabled=""
  >
    1
  </button>
  <button
    class="c3"
  >
    2
  </button>
  <button
    class="c3"
  >
    3
  </button>
  <button
    class="c1"
  >
    Next
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-right fa-w-8 "
      data-icon="angle-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-left: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination Default variant with medium size should match snapshot 1`] = `
.c1 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
}

.c1 b,
.c1 strong {
  font-weight: 800;
}

.c2 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 0.4rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c2[disabled] {
  cursor: initial;
}

.c2:first-child {
  margin-left: 0;
}

.c2:last-child {
  margin-right: 0;
}

.c2 b,
.c2 strong {
  font-weight: 800;
}

.c3 {
  background-color: #F1F2F5;
  border: none;
  color: #001254;
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  border-radius: 0.4rem;
}

.c3[disabled] {
  cursor: initial;
}

.c3:first-child {
  margin-left: 0;
}

.c3:last-child {
  margin-right: 0;
}

.c3 b,
.c3 strong {
  font-weight: 800;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="c0"
>
  <button
    class="c1 hidden"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-left fa-w-8 "
      data-icon="angle-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        fill="currentColor"
      />
    </svg>
    Previous
  </button>
  <button
    class="c2"
    disabled=""
  >
    1
  </button>
  <button
    class="c3"
  >
    2
  </button>
  <button
    class="c3"
  >
    3
  </button>
  <button
    class="c1"
  >
    Next
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-right fa-w-8 "
      data-icon="angle-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-left: 0.8rem;"
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
